spec/category_spec.rb

```diff
@@ -828,6 +828,11 @@
         expect(@dv[:yoda]).to eq(666)
       end
 
+      it "assigns at a new index" do
+        @dv.send(:set,:xtbr,3)
+        expect(@dv[:xtbr]).to eq(3)
+      end
+
       it "assigns correctly for a mixed index Vector" do
```

Comment:
You should add a spec that tests `set` correctly adds a new category. Here you are testing the existing category only.

Reply:
According to your previous suggestion of raising an error in case of an invalid category, `set` is not supposed to add only new indices.

Reply:
You can do the following:
```ruby
@dv.add_category :abc
@dv.send(:set, :xtbr, :abc)
```

Reply:
done :+1: can you please check